### PR TITLE
Automatically resolve Cloud SDK path when none is provided

### LINKED ISSUE #48: Automatically resolve Cloud SDK path when none is provided
CloudSdk class should use the PathResolver to automatically find the Cloud SDK path when one is not provided.

### PR DESCRIPTION
**Pull request by meltsufin**
Fixes #48.

@loosebazooka PTAL

### CODE REVIEW

File: src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/PathResolver.java [hunk 1]
```diff
@@ -30,7 +29,7 @@
    *
    * @return Path to Google Cloud SDK or null
    */
-  public Path getCloudSdkPath() throws FileNotFoundException {
+  public Path getCloudSdkPath() {
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/PathResolver.java]: Should we expose this as a File return given that it's public API? We can also just do an overhaul of this whole class in another PR.

File: src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/CloudSdk.java [hunk 2]
```diff
@@ -43,20 +43,44 @@
...
+   *                    the SDK in known install locations.
    */
-  public CloudSdk(File sdkPath, ProcessRunner processRunner) {
-    if (sdkPath == null) {
-      throw new NullPointerException("sdkPath cannot be null - use PathResolver for defaults");
+  public CloudSdk(File sdkPathFile, ProcessRunner processRunner) {
+    if (sdkPathFile == null) {
+      Path discoveredSdkPath = PathResolver.INSTANCE.getCloudSdkPath();
+      if (discoveredSdkPath == null) {
+        throw new NullPointerException("Google Cloud SDK path was not provided and could not be"
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/CloudSdk.java]: Is this really a NullPointerException? It seems like a more library specific exception. AppEngineException? or something else?

File: src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/CloudSdk.java [hunk 3]
```diff
@@ -43,20 +43,44 @@
   private Path sdkPath = null;
   private ProcessRunner processRunner = null;
 
-  public CloudSdk(File sdkPath) {
-    this(sdkPath, new DefaultProcessRunner());
+  /**
+   * Initializes an instance using the default location of Cloud SDK and default ProcessRunner.
+   */
+  public CloudSdk() {
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/CloudSdk.java]: Instead of all these constructors should we just provide a builder?

Comment (meltsufin) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/CloudSdk.java]: Yes, good idea, but let's do that in a separate PR.

File: src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/CloudSdk.java (hunk 2 above)
Comment (meltsufin) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/CloudSdk.java]: Agreed. Will change.

File: src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/PathResolver.java (hunk 1 above)
Comment (meltsufin) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/PathResolver.java]: With the changes to the CloudSdk class, the clients should not need to use this class at all.

Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/PathResolver.java]: agreed.

File: src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/CloudSdk.java (hunk 3 above)
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/CloudSdk.java]: Can you add a bug to track it?

Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/CloudSdk.java]: oh I see it.

**Comment by loosebazooka** [comment]
lgtm

File: src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/CloudSdk.java (hunk 3 above)
Comment (meltsufin) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/CloudSdk.java]: #51 :+1:
